src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/common/concat_transpose_sdp_transpose.cpp

```diff
@@ -299,6 +379,39 @@ TEST_P(ConcatSDPTransposeTest, CompareWithRefs) {
     auto expectedOutputs = run_test(functionRefs);
     CheckNumberOfNodesWithType(compiledModel, "ScaledDotProductAttention", 0);
     for (size_t i = 0; i < actualOutputs.size(); i++) {
+        if (i == (actualOutputs.size() - 2) && keyGroupSize > 0) {
+            const auto& key_shape = expectedOutputs[i].get_shape();
+            const auto& strides = expectedOutputs[i].get_strides();
+            auto* src = expectedOutputs[i].data<float>();
+            size_t B = key_shape[0];
+            size_t L = key_shape[1];
+            size_t H = key_shape[2];
+            size_t S = key_shape[3];
+            std::vector<uint8_t> dst_u8(B * L * H * S, 0);
+            size_t groupNum = ov::intel_cpu::div_up(L, keyGroupSize);
+            std::vector<float> scales(B * groupNum * H * S, 0);
+            std::vector<float> zp(B * groupNum * H * S, 0);
+            for (size_t b = 0; b < B; b++) {
+                for (size_t groupIdx = 0; groupIdx < groupNum; groupIdx++) {
+                    quant_u8_by_channel(src + (b * strides[0] + groupIdx * keyGroupSize * strides[1]) / expectedOutputs[i].get_element_type().size(),
+                                        dst_u8.data(),
+                                        std::min(keyGroupSize, L - groupIdx * keyGroupSize),
+                                        H * S,
+                                        strides[1] / expectedOutputs[i].get_element_type().size(),
+                                        H * S,
+                                        scales.data() + b * groupNum * H * S + groupIdx * H * S,
+                                        zp.data() + b * groupNum * H * S + groupIdx * H * S);
+                    attn_dequant_u8_by_channel(dst_u8.data(),
+                                               src + (b * strides[0] + groupIdx * keyGroupSize * strides[1]) / expectedOutputs[i].get_element_type().size(),
+                                               std::min(keyGroupSize, L - groupIdx * keyGroupSize),
+                                               H * S,
+                                               H * S,
+                                               strides[1] / expectedOutputs[i].get_element_type().size(),
+                                               scales.data() + b * groupNum * H * S + groupIdx * H * S,
+                                               zp.data() + b * groupNum * H * S + groupIdx * H * S);
+                }
+            }
+        }
```

Comment:
Why we need this for qunt_by_channel? Haven't we verified by_hidden u8 cache w/o that?

Reply:
The reference data is calculated directly using float/half-float precision. The actual data got from the state of SDPA is dequantized from u8. 

- For by_hidden u8 cases, those dequantized data could pass the elementwise comparison
- For by_channel quantization, the dequantized data cannot directly match with float/half-float reference element-wisely, but if we quantize and dequantize the reference data, they could match.

Reply:
So cannot it be solved with different thresholds for elementwise comparison? No matter how we do quantization (by channel or by hidden) resulting data should be close enough to floating point reference (with some threshold).

Reply:
Let me try threshold, thanks